File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/IDValueTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class IDValueTyped {
    public IDForm iD;
    public int value;

    public IDValueTyped(IDForm iD, int value) {
        this.iD = iD;
        this.value = value;
    }

    public IDValueTyped(IDValueString s) {
        this.iD = new IDForm(s.iD);
        this.value = Integer.parseInt(s.value);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof IDValueTyped)) return false;
        IDValueTyped that = (IDValueTyped) o;
        return Objects.equals(iD, that.iD)
            && Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(iD, value);
    }

    public IDValueString toIDValueString() {
        return new IDValueString(String.valueOf(iD), String.valueOf(value));
    }

    public static List<IDValueTyped> fromStringList(List<IDValueString> list) {
        List<IDValueTyped> result = new ArrayList<>();
        for (IDValueString s : list) result.add(new IDValueTyped(s));
        return result;
    }

    public static List<IDValueString> toStringList(List<IDValueTyped> list) {
        List<IDValueString> result = new ArrayList<>();
        for (IDValueTyped t : list) result.add(t.toIDValueString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("iD", iD == null ? null : iD.value);
        m.put("value", value);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static IDValueTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new IDValueTyped(
            new IDForm(Json.asString(Json.require(m, "iD"), "iD")),
            Json.asInt(Json.require(m, "value"), "value"));
    }

    public static IDValueTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<IDValueTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (IDValueTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<IDValueTyped> fromJSONList(String json) {
        List<IDValueTyped> result = new ArrayList<IDValueTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "IDValueTyped")));
        return result;
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/common/IDValueTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class IDValueTyped
    {
        public IDForm iD;
        public int value;

        public IDValueTyped(IDForm iD, int value)
        {
            this.iD = iD;
            this.value = value;
        }

        public IDValueString ToIDValueString()
        {
            return new IDValueString(
                Json.ToText(this.iD),
                Json.ToText(this.value)
            );
        }

        public static List<IDValueString> ToStringList(List<IDValueTyped> list)
        {
            var result = new List<IDValueString>();
            foreach (var t in list) result.Add(t.ToIDValueString());
            return result;
        }

        public static List<IDValueTyped> FromStringList(List<IDValueString> list)
        {
            var result = new List<IDValueTyped>();
            foreach (var s in list) result.Add(s.ToIDValueTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteString("iD", Json.ToText(this.iD));
            w.WriteNumber("value", this.value);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static IDValueTyped FromJsonElement(JsonElement m)
        {
            return new IDValueTyped(
                new IDForm(Json.AsString(Json.Require(m, "iD"), "iD")),
                Json.AsInt(Json.Require(m, "value"), "value")
            );
        }

        public static IDValueTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<IDValueTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<IDValueTyped> FromJSONList(string json)
        {
            var result = new List<IDValueTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "IDValueTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"ID={iD}, Value={value}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not IDValueTyped other) return false;
            return object.Equals(this.iD, other.iD)
                && object.Equals(this.value, other.value);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.iD);
            h.Add(this.value);
            return h.ToHashCode();
        }
    }
}
